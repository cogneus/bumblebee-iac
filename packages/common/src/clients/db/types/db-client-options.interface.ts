import { TableConfig } from './table-config.interface'

export interface DBClientOptions {
  region: string
  maxAttempts: number
  requestTimeout: number
  tables: Record<string, TableConfig>
}
