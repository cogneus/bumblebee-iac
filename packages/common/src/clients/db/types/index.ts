export * from './db-client.interface'
export * from './paged-items.interface'
export * from './db-client-options.interface'
export * from './table-config.interface'
export * from './transact-item.interface'
