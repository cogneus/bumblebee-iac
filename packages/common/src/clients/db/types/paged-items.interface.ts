export interface PagedItems<T> {
  items: T[]
  cursor?: string
}
