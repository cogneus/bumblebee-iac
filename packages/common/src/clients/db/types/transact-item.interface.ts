import { PutCommandInput, UpdateCommandInput, DeleteCommandInput } from '@aws-sdk/lib-dynamodb'
import { ConditionCheck } from '@aws-sdk/client-dynamodb'

export interface TransactItem {
  Put?: PutCommandInput
  Update?: UpdateCommandInput
  Delete?: DeleteCommandInput
  ConditionCheck?: ConditionCheck
}
