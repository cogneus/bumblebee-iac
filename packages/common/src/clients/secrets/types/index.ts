export * from './secrets-client.interface'
export * from './secrets-client-options.interface'
