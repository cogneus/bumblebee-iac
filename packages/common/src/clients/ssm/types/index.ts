export * from './sm-client.interface'
export * from './sm-client-options.interface'
