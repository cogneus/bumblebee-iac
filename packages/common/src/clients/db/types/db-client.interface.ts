import {
  PutCommandInput,
  PutCommandOutput,
  UpdateCommandInput,
  UpdateCommandOutput,
  DeleteCommandInput,
  DeleteCommandOutput,
  GetCommandInput,
  QueryCommandInput,
  TransactWriteCommandOutput,
} from '@aws-sdk/lib-dynamodb'
import { TableConfig } from './table-config.interface'
import { PagedItems } from './paged-items.interface'
import { TransactItem } from './transact-item.interface'
import { Destroyable } from '../../types'

export interface IDBClient extends Destroyable {
  readonly tables: Record<string, TableConfig>
  put: (params: PutCommandInput) => Promise<PutCommandOutput>
  update: (params: UpdateCommandInput) => Promise<UpdateCommandOutput>
  delete: (params: DeleteCommandInput) => Promise<DeleteCommandOutput>
  get: <T>(params: GetCommandInput) => Promise<T>
  query: <T>(params: QueryCommandInput, limit?: number, cursor?: string) => Promise<PagedItems<T>>
  queryAll: <T>(request: QueryCommandInput) => Promise<T[]>
  transactWrite: (transactions: TransactItem[]) => Promise<TransactWriteCommandOutput>
}
