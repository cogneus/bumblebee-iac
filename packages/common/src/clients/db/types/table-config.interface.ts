import { NamedRef } from '../../../types'

export interface TableConfig extends NamedRef {
  indices: Record<string, NamedRef>
  itemLimits: {
    defaultLimit: number
    maxLimit: number
  }
}
