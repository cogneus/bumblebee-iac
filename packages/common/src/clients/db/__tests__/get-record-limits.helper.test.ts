import { getRecordLimit } from '../get-record-limit.helper'
import { TableConfig } from '../types'

const tableConfig: TableConfig = {
  name: 'test',
  indices: {},
  itemLimits: {
    defaultLimit: 20,
    maxLimit: 100,
  },
}
describe('Get record limit tests', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })
  const limits = [0, -1, null, undefined, 20, 100, 101, 1000]
  test('should match limit', () => {
    const newLimits = limits.map((limit) => getRecordLimit(tableConfig, limit as any))
    expect(newLimits).toMatchSnapshot()
  })
})
