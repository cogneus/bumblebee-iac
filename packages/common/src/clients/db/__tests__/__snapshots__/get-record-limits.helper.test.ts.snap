// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get record limit tests should match limit 1`] = `
[
  20,
  20,
  20,
  20,
  20,
  100,
  100,
  100,
]
`;
